src/pages/Top.js

```diff
@@ -9,10 +9,23 @@ const Top = () => {
   const [permission, setPermission] = useState('');
   const [accountData, setAccountData] = useState({});
 
+  const usePrevious = (value) => {
```

Comment:
Great job! Let's move this custom hook outside of the `Top` component.

Let's also rename this hook to `usePreviousState` and the `value` argument to `state` so we understand that we're using it to mimic `prevState` in `componentDidUpdate`.